src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,11 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
 import java.util.List;
 
 public interface MachineProducer {
```

Comment:
```suggestion
public interface MachineProducer<T extends Machine> {
```
так ми не зможемо параметризувати імплементації класом String, Object, etc